code/modules/power/supermatter/supermatter_hit_procs.dm

```diff
@@ -24,8 +24,9 @@
 		if(power_changes) //This needs to be here I swear
 			power += projectile.damage * bullet_energy + kiss_power
 			if(!has_been_powered)
-				investigate_log("has been powered for the first time.", INVESTIGATE_ENGINE)
-				message_admins("[src] has been powered for the first time [ADMIN_JMP(src)].")
+				var/fired_from_exists = projectile.fired_from ? " with [projectile.fired_from]" : ""
+				investigate_log(projectile.firer ? "has been powered for the first time by [ADMIN_FULLMONTY(projectile.firer)][fired_from_exists]." : "has been powered for the first time.", INVESTIGATE_ENGINE)
+				message_admins(projectile.firer ? "[src] [ADMIN_JMP(src)] has been powered for the first time by [ADMIN_FULLMONTY(projectile.firer)][fired_from_exists]." : "[src] has been powered for the first time [ADMIN_JMP(src)].")
```

Comment:
put each part of the ternary on a new line or else

Reply:
I was just gonna say use this
```var/fired_from_exists = projectile.fired_from ? " with [projectile.fired_from]" : ""
var/shown_message = "has been powered for the first time [projectile.firer ? " by [ADMIN_FULLMONTY(projectile.firer)][fired_from_exists].": "."]"
investigate_log(shown_message , INVESTIGATE_ENGINE)
message_admins(shown_message)```